blurring_as_a_service/metrics/custom_metrics_calculator.py

```diff
@@ -0,0 +1,500 @@
+                "true_positives": 0,
+                "true_positives_small": 0,
+                "true_positives_medium": 0,
+                "true_positives_large": 0,
+                "false_negatives": 0,
+                "false_negatives_small": 0,
+                "false_negatives_medium": 0,
+                "false_negatives_large": 0,
+            }
+            for available_category in available_categories
+        }
+        for value in self._statistics_per_category.values():
+            category = all_categories[value[category_name]]
+            category["true_positives"] += value["true_positives"]
+            category["false_negatives"] += value["false_negatives"]
+            category["true_positives_small"] += value["true_positives_small"]
+            category["true_positives_medium"] += value["true_positives_medium"]
+            category["true_positives_large"] += value["true_positives_large"]
+            category["false_negatives_small"] += value["false_negatives_small"]
+            category["false_negatives_medium"] += value["false_negatives_medium"]
+            category["false_negatives_large"] += value["false_negatives_large"]
+
+        result = []
+        for name, values in all_categories.items():
+            true_positives = values["true_positives"]
+            false_negatives = values["false_negatives"]
+            true_positives_small = values["true_positives_small"]
+            true_positives_medium = values["true_positives_medium"]
+            true_positives_large = values["true_positives_large"]
+            false_negatives_small = values["false_negatives_small"]
+            false_negatives_medium = values["false_negatives_medium"]
+            false_negatives_large = values["false_negatives_large"]
+
+            false_negative_rate = (
+                self._calculate_false_negative_rate(false_negatives, true_positives)
+                if false_negatives + true_positives > 0
+                else None
+            )
+
+            false_negative_rate_small = (
+                self._calculate_false_negative_rate(
+                    false_negatives_small, true_positives_small
+                )
+                if false_negatives_small + true_positives_small > 0
+                else None
+            )
+
+            false_negative_rate_medium = (
+                self._calculate_false_negative_rate(
+                    false_negatives_medium, true_positives_medium
+                )
+                if false_negatives_medium + true_positives_medium > 0
+                else None
+            )
+
+            false_negative_rate_large = (
+                self._calculate_false_negative_rate(
+                    false_negatives_large, true_positives_large
+                )
+                if false_negatives_large + true_positives_large > 0
+                else None
+            )
+
+            result.append(
+                {
+                    "category": category_name,
```

Comment:
It could be an idea to have this as an object in the future to make everything cleaner.